src/v/cluster/archival/ntp_archiver_service.cc

```diff
@@ -2377,10 +2377,6 @@ ntp_archiver::maybe_truncate_manifest() {
           "archival metadata cleanup, some segments will be removed from the "
           "manifest, start offset before cleanup: {}",
           manifest().get_start_offset());
-        retry_chain_node rc_node(
-          _conf->manifest_upload_timeout(),
-          _conf->upload_loop_initial_backoff(),
-          &rtc);
```

Comment:
I assume this is why you put Evgeny on the PR, but I'm curious to know whether this was a bug of some kind or dead code.

Reply:
Seems like dead code from the very beginning https://github.com/redpanda-data/redpanda/commit/8f348ee061dc350f52faabe0f8eded318f9c993a

`_partition->archival_meta_stm()->truncate` is called with a deadline instead rather than a rcn.